skyportal/handlers/api/comment.py

```diff
@@ -59,32 +71,71 @@ def get(self, comment_id, associated_resource_type=None):
                 schema: Error
         """
 
-        if associated_resource_type is None:
-            associated_resource_type = 'object'
-
-        if associated_resource_type.lower() == "object":  # comment on object (default)
+        try:
+            comment_id = int(comment_id)
+        except (TypeError, ValueError):
+            return self.error("Must provide a valid comment ID. ")
+
+        # the default is to comment on an object
+        if associated_resource_type.lower() in (
+            "object",
+            "objects",
+            "source",
+            "sources",
+        ):
             comment = Comment.get_if_accessible_by(
                 comment_id, self.current_user, raise_if_none=True
             )
-        elif associated_resource_type.lower() == "spectrum":
+            comment_resource_id_str = str(comment.obj_id)
+
+            # comment.api_path = f'api/sources/{resource_id}/comment/{comment_id}'
+        elif associated_resource_type.lower() in ("spectrum", "spectra"):
             comment = CommentOnSpectrum.get_if_accessible_by(
                 comment_id, self.current_user, raise_if_none=True
             )
+            comment_resource_id_str = str(comment.spectrum_id)
+            # comment.api_path = f'api/spectrum/{resource_id}/comment/{comment_id}'
         # add more options using elif
         else:
             return self.error(
                 f'Unsupported input "{associated_resource_type}" given as "associated_resource_type" argument.'
             )
 
+        if comment_resource_id_str != resource_id:
+            return self.error(
+                f'Comment resource ID does not match resource ID given in path ({resource_id})'
+            )
         return self.success(data=comment)
 
     @permissions(['Comment'])
-    def post(self):
+    def post(self, associated_resource_type, resource_id, *_):
         """
         ---
         description: Post a comment
         tags:
           - comments
+        parameters:
+          - in: path
+            name: associated_resource_type
+            required: true
+            schema:
+              type: string
+            description: |
+               What underlying data the comment is on:
+               an "object" (default), or a "spectrum".
+          - in: path
+            name: resource_id
+            required: true
+            schema:
+              type: string
+            description: |
+               The ID of the underlying data.
+               This would be a string for an object ID
+               or an integer for other data types like spectrum.
+               The object pointed to by this input must be a valid
```

Comment:
I would make this explicit until we go more general: "This is the ID of the object, source, or spectrum the comment is posted to."